community-tutorials/how-to-install-uptime-kuma/01-en.md

```diff
@@ -0,0 +1,201 @@
+4. In the new window click on "DNS"
+5. Scroll down until there are empty fields
+6. In the first field write your subdomain-name (I recommend using "status") 
+7. In the dropdown menu select `A`
+8. In the last field write the IP of your Server (from Step 1)
+9. Click on "Save DNS records"
+
+# Step 3 - Preparing the Server
+First you need to log-in to your Server via SSH, then:
+
+1. Enter `sudo -s` to get root privileges
+2. If asked for your password enter it and press `Enter`
+3. Now enter `apt update -y && apt upgrade -y` to update the Server
+4. If a pink window appears, press `Enter` once to continue
+5. Enter `curl https://deb.nodesource.com/setup_16.x -o /tmp/nodesource_setup.sh && bash /tmp/nodesource_setup.sh` to install NodeJS 16
+6. Enter `apt install nodejs curl nginx git -y` to install all needed software
+7. If a pink window appears, press `Enter` once to continue
+8. Enter `npm install npm -g` to update npm
+9. If a pink window appears, press `Enter` once to continue
+10. Enter `npm install pm2 -g && pm2 completion install` to install pm2 and its autocompletion
+11. Enter `exit` and press `Enter` to leave root privileges
+
+# Step 4 - Installing Uptime Kuma
+Enter the following command to install Uptime Kuma:  
+1. Enter `git clone https://github.com/louislam/uptime-kuma.git` to clone the repository
+2. Change directory by executing `cd uptime-kuma`
+3. Enter `npm run setup` to install all dependencies
+4. Now start kuma by entering `npm run start-server`
+
+Once you see "Listening on 3001" in the terminal, go to your IP on Port 3001 (e.g. `123.123.123.123:3001`) in your browser.
+If you see the Uptime Kuma setup screen, you have successfully installed Uptime Kuma, please stop the server by pressing `Ctrl + C` in the terminal once and waiting shortly until the server stops.
+
+# Step 5 - Reverse Proxy with Nginx
+1. Enter `sudo -s` to get root privileges
+2. If asked for your password, enter it and press `Enter`
+3. Enter `nano /etc/nginx/sites-available/uptime-kuma` to create a new nginx config file
+4. Paste the following code into the file:
+```nginx
+server  {
+    listen 80;
+    # Remove '#' in the next line to enable IPv6
+    # listen [::]:80;
+    server_name    sub.domain.com;
+    location / {
+        proxy_pass         http://localhost:3001;
+        proxy_http_version 1.1;
+        proxy_set_header   Upgrade $http_upgrade;
+        proxy_set_header   Connection "upgrade";
+        proxy_set_header   Host $host;
+    }
+}
+```
+5. Replace `sub.domain.com` with your subdomain (e.g. `status.example.com`)
+6. Exit the file by pressing `Ctrl + X` and then `Y` and then `Enter`
+7. Enter `ln -s /etc/nginx/sites-available/uptime-kuma /etc/nginx/sites-enabled/uptime-kuma` to enable the config file
+8. Enter `nginx -t` to test the config file
+9. If you see `test is successful` you can reload nginx by entering `systemctl reload nginx`
+
+# Step 6 - Setting up Uptime Kuma to start on boot
```

Comment:
Shouldn't "pm" be used for this? I also installed Kuma and used pm as in the instructions on https://github.com/louislam/uptime-kuma/ and Kuma starts automatically for me.

Reply:
I really dislike using software that is just using a system-wide-automation-program and also is usable for one use-case (Node.js) and also just adds another point of failure. In my installation it didn't work properly, but using systemctl always works. Also pm2 is just using systemctl under the hood. Thats why I chose to use systemctl rather than pm2.

Reply:
Dear contributors, thank you two for your input. We are currently reviewing the tutorial to ensure technical correctness and will consider your comments as part of this review process. Personally, I can see why someone might want to got for pm2 in this case, especially when someone might be unwilling to rely on systemd for philosophical or other reasons, but for this tutorial using Ubuntu as the distribution, utilicing systemctl for the automation seems like a great solution.